tests/e2e/json_writer.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+)
+
+// TraceWriter is an interface for writers that write steps to files.
```

Comment:
Do we need a comment on whether this would override the file with the steps, append the steps, etc.?

Reply:
Sure, doesn't hurt I think
b4e35e35127bb3db0af7f780c72e05c37642be67